### DISCUSSION TITLE
should kafka instrumentation support `mq.broker` attribute?

### DISCUSSION THREAD

**Discussion by huange7** [open-telemetry/opentelemetry-java-instrumentation#11051] [Q&A] [answered]
The same attribute has appeared in RocketMQ, for example, `messaging.rocketmq.broker_address`. As far as I understand, Kafka also has broker address, and so do RabbitMQ and ActiveMQ. Should we create an `AttributesGetter` to support this attribute?

**Comment by laurit** [comment, marked as answer]
https://github.com/open-telemetry/semantic-conventions/blob/main/docs/messaging/messaging-spans.md suggests that `server.address` should be used for the broker.

**Reply by huange7** [reply]
Thank you for your sharing, but I noticed that the Kafka instrumentation does not inject `server.address`. What is the reason for not considering it?

**Reply by laurit** [reply]
In instrumentation code all the useful information is not readily available, perhaps getting the value for `server.address` is hard and nobody has implemented it.

**Reply by huange7** [reply]
I understand now. May I convert this into an issue?

**Reply by laurit** [reply]
There already is https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/10647

**Reply by huange7** [reply]
oh, thanks!
